src/snippet.rs

```diff
@@ -38,104 +35,178 @@ impl<'a> Message<'a> {
         self
     }
 
-    pub fn snippet(mut self, slice: Snippet<'a>) -> Self {
-        self.snippets.push(slice);
+    pub fn group(mut self, group: Group<'a>) -> Self {
```

Comment:
One of the places where we deal with groups, we'll need to explain the difference between a single group vs many groups.  Unsure whether a suggested policy belongs in this package or somewhere else but at least Cargo and Rustc will need to be consistent